solutions/src/main/java/ru/sberbank/school/task02/ExConverter.java

```diff
@@ -0,0 +1,110 @@
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(@NonNull ClientOperation operation,
+                                                @NonNull Symbol symbol,
+                                                @NonNull BigDecimal amount,
+                                                         double delta,
+                                                @NonNull Beneficiary beneficiary) {
+        return convertReversed(operation,symbol,amount,beneficiary);
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(@NonNull ClientOperation operation,
+                                                @NonNull Symbol symbol,
+                                                @NonNull BigDecimal amount,
+                                                @NonNull Beneficiary beneficiary) {
+        if (amount.compareTo(BigDecimal.ZERO) <= 0) {
+            throw new IllegalArgumentException("Amount can't be lower or equals 0");
+        }
+
+        List<Quote> quotes = service.getQuotes(symbol);
+        if (quotes.isEmpty()) {
+            return Optional.empty();
+        }
+        BigDecimal tempCount;
+        Quote resultQuote = null;
+        Quote lastQuote = null;
+
+        for (Quote q : quotes) {
+            tempCount = getCount(q,amount,operation);
+
+            if (quoteIsMatch(q, tempCount) && (lastQuote == null || compareQuotes(lastQuote,q,operation,amount))) {
+                resultQuote = chooseBenef(lastQuote, q, operation,beneficiary);
+                lastQuote = q;
+            }
+        }
+
+        if (resultQuote == null) {
+            return Optional.empty();
+        }
+
+        return Optional.of(BigDecimal.ONE.divide(getPrice(resultQuote,operation),
+                                            10,
+                                                RoundingMode.HALF_UP));
+
+    }
+
+    private boolean quoteIsMatch(Quote quote, BigDecimal amount) {
+        return (quote.getVolumeSize().compareTo(amount) > 0);
+    }
+
+    private BigDecimal getPrice(Quote quote, ClientOperation operation) {
+        return operation == ClientOperation.BUY ? quote.getOffer() : quote.getBid();
+    }
+
+
+    private BigDecimal getCount(Quote quote, BigDecimal amount, ClientOperation operation) {
+        return operation == ClientOperation.BUY
+                ? amount.divide(quote.getOffer(),
+                10,
+                RoundingMode.HALF_UP)
+                : amount.divide(quote.getBid(),
+                10,
+                RoundingMode.HALF_UP);
+    }
+
+    private boolean compareQuotes(Quote lastQuote, Quote candidateQuote, ClientOperation operation, BigDecimal amount) {
+        return getCount(candidateQuote, amount, operation).compareTo(lastQuote.getVolumeSize()) > 0;
+    }
+
+    private Quote chooseBenef(Quote q1, Quote q2, ClientOperation operation, Beneficiary beneficiary) {
+        if(q1 == null) {
+            return q2;
+        }
+        if(q2 == null) {
```

Comment:
WhitespaceAround: &#39;if&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-84386135>*